yoyodyne/predict.py

```diff
@@ -23,51 +22,35 @@ def get_trainer_from_argparse_args(
     return pl.Trainer.from_argparse_args(args, max_epochs=0)
 
 
-def get_dataset_from_argparse_args(
+def get_datamodule_from_argparse_args(
     args: argparse.Namespace,
-) -> datasets.BaseDataset:
+) -> data.DataModule:
     """Creates the dataset from CLI arguments.
 
     Args:
         args (argparse.Namespace).
 
     Returns:
-        datasets.BaseDataset.
+        data.DataModule.
     """
-    config = dataconfig.DataConfig.from_argparse_args(args)
-    return datasets.get_dataset(args.predict, config, args.index)
-
-
-def get_loader(
-    dataset: datasets.BaseDataset,
-    arch: str,
-    batch_size: int,
-    max_source_length: int,
-    max_target_length: int,
-) -> data.DataLoader:
-    """Creates the loader.
-
-    Args:
-        dataset (data.Dataset).
-        arch (str).
-        batch_size (int).
-        max_source_length (int).
-        max_target_length (int).
-
-    Returns:
-        data.DataLoader.
-    """
-    collator = collators.Collator(
-        dataset,
-        arch,
-        max_source_length,
-        max_target_length,
-    )
-    return data.DataLoader(
-        dataset,
-        collate_fn=collator,
-        batch_size=batch_size,
-        num_workers=1,
+    separate_features = args.features_col != 0 and args.arch in [
+        "pointer_generator_lstm",
+        "transducer",
+    ]
+    # TODO(kbg): reuse index?
```

Comment:
If I understand correctly, yes. We need to reuse the index from training for this to work.

Reply:
Done.